Assets/SEE/Controls/Actions/MarkAction.cs

```diff
@@ -0,0 +1,164 @@
+                result = true;
+                CurrentState = IReversibleAction.Progress.Completed;
+                AudioManagerImpl.EnqueueSoundEffect(IAudioManager.SoundEffect.NewNodeSound, parent);
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// The node that was added when this action was executed. It is saved so
+        /// that it can be removed on Undo().
+        /// </summary>
+        private GameObject addedSphere;
+
+        /// <summary>
+        /// Memento capturing the data necessary to re-do this action.
+        /// </summary>
+        private Memento memento;
+
+        /// <summary>
+        /// The information we need to re-add a node whose addition was undone.
+        /// </summary>
+        private struct Memento
+        {
+            /// <summary>
+            /// The parent of the new node.
+            /// </summary>
+            public readonly GameObject Parent;
+
+            /// <summary>
+            /// The position of the new node in world space.
+            /// </summary>
+            public readonly Vector3 Position;
+
+            /// <summary>
+            /// The scale of the new node in world space.
+            /// </summary>
+            public readonly Vector3 Scale;
+
+            /// <summary>
+            /// The node ID for the added node. It must be kept to re-use the
+            /// original name of the node in Redo().
+            /// </summary>
+            public string NodeID;
+
+            /// <summary>
+            /// Constructor setting the information necessary to re-do this action.
+            /// </summary>
+            /// <param name="markedNode">The marked node</param>
+            public Memento(GameObject parent, Vector3 position, Vector3 scale)
+            {
+                Parent = parent;
+                Position = position;
+                Scale = scale;
+                NodeID = null;
+            }
+        }
+
+        /// <summary>
+        /// Returns all IDs of gameObjects manipulated by this action.
+        /// </summary>
+        /// <returns>all IDs of gameObjects manipulated by this action</returns>
+        public override HashSet<string> GetChangedObjects()
+        {
+            return new HashSet<string>
+            {
+                memento.Parent.name,
+                memento.NodeID
+            };
+        }
+
+        /// <summary>
+        /// Returns a new instance of <see cref="MarkAction"/>.
+        /// </summary>
+        /// <returns>new instance</returns>
+        public override IReversibleAction NewInstance()
+        {
+            return CreateReversibleAction();
+        }
+
+        /// <summary>
+        /// Undoes this MarkAction.
+        /// </summary>
+        public override void Undo()
+        {
+            base.Undo();
+            if (addedSphere != null)
+            {
+                new DeleteNetAction(addedSphere.name).Execute();
```

Comment:
This will not work because your sphere is neither a node nor an edge and hence will not be added to `GraphElementIDMap`.